### PR TITLE
mkdir: set correct permissions on dirs created by -p

### PR DESCRIPTION
**Pull request by shinhs0506**
should fix mkdir/perm.sh
1. default permission created by mkdir is `a=rwx` (0o0777)
2. permission of parent dirs created by -p argument is `=rwx,u+wx`, which is umask modified by `u+wx`

### CODE REVIEW

File: src/uu/mkdir/src/mkdir.rs [hunk 1]
```diff
@@ -179,7 +179,8 @@ fn chmod(_path: &Path, _mode: u32) -> UResult<()> {
     Ok(())
 }
 
-fn create_dir(path: &Path, recursive: bool, verbose: bool) -> UResult<()> {
+#[allow(unused_variables)]
```
Comment (sylvestre) [inline src/uu/mkdir/src/mkdir.rs:182]: could you please document why this allow? thanks

**Comment by sylvestre** [comment]
> GNU testsuite comparison:
> 
> ```
> Congrats! The gnu test tests/mkdir/perm is no longer failing!
> ```

bravo :)

**Comment by cakebaker** [comment]
The two new tests (`test_mkdir_parent_mode` and `test_mkdir_parent_mode_check_existing_parent`) pass, even though I didn't apply any of the changes to `mkdir.rs`. Is that intentional?

**Comment by shinhs0506** [comment]
Oops, I updated the tests, they should fail now. Thanks for catching that

**Comment by sylvestre** [comment]
thanks :)

**Comment by niyaznigmatullin** [comment]
I ran the test locally, and the test `test_symbolic_alteration` fails after this change. I think it does because my umask is 0002, which is default in my terminal, but if umask is set to 0022, then it starts working correctly. We already discussed umask issues here #3645. In my opinion, that kind of tests have to set umask explicitly before running `mkdir -m`, so that the tests work in every terminal window, no matter what umask is set there.

**Comment by shinhs0506** [comment]
@niyaznigmatullin The issue might be caused by the tests I added. Could you try running 
`cargo test --features "mkdir" --no-default-features -- --test-threads=1` 
and see if the error goes away?

**Pull request merged**